### PR TITLE
Fix the error of hostAliases when there are more than 2 hostnames

### PR DESCRIPTION
**Pull request by cdmikechen**
If there are more than 2 hostnames in `hostAliases` path, `spec/hostAliases` will only show the first and many last hostname. Like this:
```yaml
hostAliases:
- ip: "10.1.1.1"
  hostnames:
  - "hostname1"
- ip: "10.1.1.3"
  hostnames:
  - "hostname3"
- ip: "10.1.1.3"
  hostnames:
  - "hostname3"
```
Now fix this problem.

### CODE REVIEW

**Comment by cdmikechen** [comment]
@googlebot I signed it!

File: pkg/webhook/patch.go [hunk 1]
```diff
@@ -795,19 +795,9 @@ func addHostAliases(pod *corev1.Pod, app *v1beta2.SparkApplication) []patchOpera
...
 	var ops []patchOperation
-	for _, v := range hostAliases {
-		if first {
-			ops = append(ops, patchOperation{Op: "add", Path: "/spec/hostAliases", Value: []corev1.HostAlias{v}})
-			first = false
-		} else {
-			ops = append(ops, patchOperation{Op: "add", Path: "/spec/hostAliases/-", Value: &v})
-		}
+	if len(hostAliases) > 0 {
+		ops = append(ops, patchOperation{Op: "add", Path: "/spec/hostAliases", Value: hostAliases})
```
Comment (liyinan926) [inline pkg/webhook/patch.go:800]: If the pod already has some host alias, we should use `/spec/hostAliases/-` for the path.

**Comment by cdmikechen** [comment]
@googlebot I fixed it.

**Comment by cdmikechen** [comment]
@liyinan926  Hi~ Thanks for your review.
I've changed my code, and run test in my computer. But I've seen CI build failed. Can you help me find the cause of the specific error, please ? I haven't found the connection between my code and the CI build failed.

**Comment by cdmikechen** [comment]
@liyinan926 Hi~ CI have passed by PR https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/1210
I've merge this change to codes.

**Review by liyinan926 — APPROVED** [review]
LGTM

**Pull request merged**